portal/app/api/v1/endpoint/lora_device.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework.authentication import BasicAuthentication
+from app.api.v1 import CsrfExemptAuth
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from app.models import FarmDevice
+import json
+
+
+class LoraDeviceViewSet(APIView):
+    authentication_classes = (CsrfExemptAuth.CsrfExemptSessionAuthentication, BasicAuthentication)
+
+    def dataparsing(self, data):
+        try:
+            temperature = data['event_data']['payload'][2]['value']
+            rel_humidity = data['event_data']['payload'][3]['value']
+            batt_value = data['event_data']['payload'][4]['value']
+            timestamp = data['event_data']['timestamp']
+            dev_internal_id = data['device']['id']
+            device_uid = data['device']['thing_name'][1:3]
+            return [device_uid, json.dumps({'temperature': temperature, 'rel_humidity': rel_humidity, 'batt_value': batt_value,
+                           'dev_internal_id': dev_internal_id,'timestamp': timestamp})]
+        except Exception as e:
+            print('Error in Lora dataparsing: ', e)
+            return None
+
+    def post(self, request):
```

Comment:
I was thinking something more similar to this: https://www.django-rest-framework.org/api-guide/views/#api_view

But I guess it's fine.

Reply:
Ok! I thought about it but didn't implement will do it now.